### ISSUE TITLE
Frida modules aren't found by their full path on iOS

### ISSUE THREAD

**Issue by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#547]
and here i'm :-)

everything is up to date, the message is:

Spawned `com.nintendo.zaka`. Resuming main thread!                      
[iOS Device::com.nintendo.zaka ]-> il2cpp: UnityFramework has been loaded, but such event hasn't been detected - please file a bug

**Comment by vfsfitvnm** [comment]
Thanks for reporting! Unfortunately I do not have a iOS device so I cannot fix it. Would you be able to take a stab at it?

The problem is we couldn't catch the moment when the IL2CPP module (named `UnityFramework` on iOS)  was loaded: https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/97ffc512a8a73886fa413716886327c3546c8134/src/utils/native-wait.ts#L68-L75

Currently, on iOS the following exports are intercepted:
https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/97ffc512a8a73886fa413716886327c3546c8134/src/utils/native-wait.ts#L47-L49

It looks like these two exports did not load `UnityFramework`.

Relevant files: 
https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/module.ts
https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/utils/native-wait.ts

I'd start by executing the following Frida script

```ts
const handle = Module.getExportByName(null, "dlopen");
Interceptor.attach(handle, (args) => console.log(`dlopen: loading ${args[0].readUtf8String()}`));
```
to confirm `UnityFramework` isn't loaded by `dlopen`.

**Comment by SamueleD78** [comment]
so that's the output:

dlopen: loading /private/var/containers/Bundle/Application/24D31E44-A0E7-40EE-A36C-2EA7818522E5/zaka.app/Frameworks/UnityFramework.framework/UnityFramework

what can i do more to help?

**Comment by vfsfitvnm** [comment]
That's weird.
Another thing that could happen is that `UnityFramework` gets loaded after we checked for its availability, but before we attach the interceptors. Is this bug reproducible 100% of the times?

We could add a very simple piece of code to cover it. Would you run this Frida script?
https://gist.github.com/vfsfitvnm/b0bc4080dc729eb2ec34df0fbffc2b4c

**Comment by SamueleD78** [comment]
that's the output:

Spawned `com.nintendo.zaka`. Resuming main thread!                      
[iOS Device::com.nintendo.zaka ]-> il2cpp: UnityFramework has been loaded, but such event hasn't been detected - please file a bug
Ready! Did it work?

yes it's 100% of the times

**Comment by vfsfitvnm** [comment]
I updated the gist to add some logging ^^

**Comment by SamueleD78** [comment]
Spawning `com.nintendo.zaka`...                                         
issue#547: found export dlopen@libdyld.dylib at 0x18615a518
issue#547: 1 target(s) found: [object Object]
issue#547: timeout installed
issue#547: interceptor installed
issue#547: waiting...
Spawned `com.nintendo.zaka`. Resuming main thread!                      
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation'
[iOS Device::com.nintendo.zaka ]-> issue#547: target at 0x18615a518 called with '/usr/lib/substrate/SubstrateLoader.dylib'
issue#547: target at 0x18615a518 called with '/usr/sbin/aslmanager'
issue#547: target at 0x18615a518 called with '/usr/lib/system/libsystem_sandbox.dylib'
issue#547: target at 0x18615a518 called with '/usr/lib/substrate/SubstrateInserter.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/Foundation.framework/Foundation'
issue#547: target at 0x18615a518 called with '/var/containers/Bundle/Application/24D31E44-A0E7-40EE-A36C-2EA7818522E5/zaka.app/Frameworks/UnityFramework.framework/UnityFramework'
issue#547: target at 0x18615a518, called with '/var/containers/Bundle/Application/24D31E44-A0E7-40EE-A36C-2EA7818522E5/zaka.app/Frameworks/UnityFramework.framework/UnityFramework', loaded module 'UnityFramework'
issue#547: module 'UnityFramework' instance is null (alternative is: null)
issue#547: target at 0x18615a518 called with '/var/containers/Bundle/Application/24D31E44-A0E7-40EE-A36C-2EA7818522E5/zaka.app/Frameworks/UnityFramework.framework/UnityFramework'
issue#547: target at 0x18615a518 called with '/usr/lib/libobjc.A.dylib'
issue#547: target at 0x18615a518 called with '/usr/lib/libobjc-trampolines.dylib'
issue#547: target at 0x18615a518, called with '/var/containers/Bundle/Application/24D31E44-A0E7-40EE-A36C-2EA7818522E5/zaka.app/Frameworks/UnityFramework.framework/UnityFramework', loaded module 'UnityFramework'
issue#547: module 'UnityFramework' instance is null (alternative is: [object Object])
issue#547: target at 0x18615a518 called with '/usr/lib/libboringssl.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/PrivateFrameworks/CoreServicesInternal.framework/CoreServicesInternal'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/MobileCoreServices.framework/MobileCoreServices'
issue#547: target at 0x18615a518 called with '/System/Library/PrivateFrameworks/ColorSync.framework/ColorSync'
issue#547: target at 0x18615a518 called with ''
issue#547: target at 0x18615a518 called with '/System/Library/Extensions/AGXMetalA10.bundle/AGXMetalA10'
issue#547: target at 0x18615a518 called with '/System/Library/Extensions/AGXMetalA10.bundle/AGXMetalA10'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/UIKit.framework/UIKit'
issue#547: target at 0x18615a518 called with '/System/Library/Extensions/IOHIDFamily.kext/PlugIns/IOHIDLib.plugin/IOHIDLib'
issue#547: target at 0x18615a518 called with '/System/Library/Extensions/IOHIDFamily.kext/PlugIns/IOHIDLib.plugin/IOHIDLib'
issue#547: target at 0x18615a518 called with '/usr/lib/libRosetta.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/PrivateFrameworks/UserManagement.framework/UserManagement'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/MobileCoreServices.framework/MobileCoreServices'
issue#547: target at 0x18615a518 called with '/System/Library/PrivateFrameworks/GPUCompiler.framework/Libraries/libmetal_timestamp.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/UIKit.framework/UIKit'
issue#547: target at 0x18615a518 called with '/System/Library/PrivateFrameworks/CloudDocs.framework/CloudDocs'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/MediaToolbox.framework/MediaToolbox'
issue#547: target at 0x18615a518 called with '/usr/lib/swift/libswiftCore.dylib'
issue#547: target at 0x18615a518 called with '/usr/lib/libnetwork.dylib'
issue#547: target at 0x18615a518 called with '/usr/lib/libusrtcp.dylib'
issue#547: target at 0x18615a518 called with '/usr/lib/libquic.dylib'
issue#547: target at 0x18615a518 called with '/usr/lib/libnetwork.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/CFNetwork.framework/CFNetwork'
issue#547: target at 0x18615a518 called with '/usr/lib/log/liblog_network.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/Accelerate.framework/Frameworks/vImage.framework/Libraries/libCGInterfaces.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/PrivateFrameworks/CoreMaterial.framework/CoreMaterial'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/AudioToolbox.framework/libEmbeddedSystemAUs.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/AudioToolbox.framework/libAudioDSP.dylib'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/AudioToolbox.framework/AudioCodecs'
issue#547: target at 0x18615a518 called with '/System/Library/CoreServices/RawCamera.bundle/RawCamera'
issue#547: target at 0x18615a518 called with '/System/Library/Frameworks/MediaToolbox.framework/MediaToolbox'
il2cpp: UnityFramework has been loaded, but such event hasn't been detected - please file a bug
Ready! Did it work?

**Comment by vfsfitvnm** [comment]
```
issue#547: target at 0x18615a518, called with '/var/containers/Bundle/Application/24D31E44-A0E7-40EE-A36C-2EA7818522E5/zaka.app/Frameworks/UnityFramework.framework/UnityFramework', loaded module 'UnityFramework'
issue#547: module 'UnityFramework' instance is null (alternative is: [object Object])
```

Now I have fix, thanks!

**Comment by vfsfitvnm** [comment]
Fixed in v0.9.1. Thanks :)

**Comment by SamueleD78** [comment]
thanks to you!! ^^ your response was fast and nice!
